_posts/2019/2019-09-22-a-rails-active-model-validator-for-stripe-ids.md

```diff
@@ -0,0 +1,83 @@
+---
+title: "A Active Model validator for Stripe Ids"
+description: "A decent example"
+layout: article
+category: ruby
+image:
+  base: '2019/a-rails-active-model-validator-for-stripe-ids'
+  alt: 'Zebra stripes'
+  credit: David Clarke
+  source: "https://unsplash.com/photos/sN6d60TySV0"
+
+---
+
+In [my article on using validators](/ruby/use-a-custom-validator), the example was pretty simple.
+
+This is an example of real production use of a more complex validator extracted from the billing code in [AnswerThePublic](https://answerthepublic.com). We use [Stripe](https://stripe.com) exclusively for billing in our applications.
+
+Every object on Stripe—a transaction, an invoice, a customer—has a unique and similarly-patterned identifier. We often need to store those `id`s locally in our application to interact with Stripe so we can collect payments from customers.
```

Comment:
*these `id`s

*so that they can interact with Stripe and we can collect payments...